pkg/controller/v1beta1/inferenceservice/utils/utils.go

```diff
@@ -258,7 +258,10 @@ func UpdateImageTag(container *v1.Container, runtimeVersion *string, servingRunt
 				// For TFServing/TorchServe the GPU image is tagged with suffix "-gpu", when the version is found in the tag
 				// and runtimeVersion is not specified, we default to append the "-gpu" suffix to the image tag
 				if servingRuntime != nil && (*servingRuntime == constants.TFServing || *servingRuntime == constants.TorchServe) {
-					container.Image = image + "-gpu"
+					//check for the case when image field is specified directly with gpu tag
+					if !strings.HasSuffix(container.Image, "-gpu") {
+						container.Image = image + "-gpu"
```

Comment:
Why do we need to add `-gpu` tag and not just directly pick the image field as it is ? 
Like this:
`container.Image = image`

Reply:
The torchserve cpu and gpu image has different tags, on the serving runtime it defines the cpu image tag so we need to append the gpu suffix based on the resource requirement user defined on inference service yaml.

Reply:
Got it.